tests/extra/cold.ts

```diff
@@ -0,0 +1,58 @@
+/// <reference path="../../typings/globals/mocha/index.d.ts" />
+/// <reference path="../../typings/globals/node/index.d.ts" />
+import xs from '../../src/index';
+import cold from '../../src/extra/cold';
+import * as assert from 'assert';
+
+describe('cold (extra)', () => {
+  it('should push values to a listener', (done) => {
+    const stream = xs.periodic(100).take(3).compose(cold);
+
+    const expected = [0, 1, 2];
+    const listener = {
+      next (x: number) {
+        assert(x === expected.shift());
+      },
+      error: done,
+      complete () {
+        assert(expected.length === 0);
+        stream.removeListener(listener);
+        done();
+      }
+    };
+
+    stream.addListener(listener);
+  });
+
+  it('should restart the producer for every new listener', (done) => {
+    const stream = xs.periodic(100).take(3).compose(cold);
```

Comment:
And... that may hurt reasonaboutability of xstream in general, because it's easier if we can assume that "*all streams are hot*" than to assume that "*it may be cold or hot, we don't know*".

I'm not sure what to do in this situation, but I'd like to explore first how this could be solved elegantly (no Promise resolve for instance) with xstream as it is.